### PR TITLE
Add test to make sure member_presenters gets the right class

### PR DESCRIPTION
**Pull request by hackmastera**
Fails as below:

> .............F
>
> Failures:
> 
>   1) Sufia::WorkShowPresenter#member_presenters returns appropriate classes for each
>      Failure/Error: expect(presenter.member_presenters.size).to eq 1
>      
>      RSolr::Error::Http:
>        RSolr::Error::Http - 400 Bad Request
>        Error: 'org.apache.solr.search.SyntaxError: Cannot parse \'proxy_in_ssi:\': Encountered "<EOF>" at line 1, column 13.
>      
>        URI: http://127.0.0.1:8985/solr/hydra-test/select?wt=ruby&fl=ordered_targets_ssim&q=proxy_in_ssi%3A&qt=standard
>      
>        Backtrace: /usr/local/lib/ruby/gems/2.2.0/gems/rsolr-1.1.2/lib/rsolr/client.rb:288:in `adapt_response'
>        /usr/local/lib/ruby/gems/2.2.0/gems/rsolr-1.1.2/lib/rsolr/client.rb:189:in `execute'
>        /usr/local/lib/ruby/gems/2.2.0/gems/rsolr-1.1.2/lib/rsolr/client.rb:175:in `send_and_receive'
>        (eval):2:in `get'
>        /usr/local/lib/ruby/gems/2.2.0/gems/active-fedora-11.0.0.rc7/lib/active_fedora/solr_service.rb:41:in `get'
>        /usr/local/lib/ruby/gems/2.2.0/gems/active-fedora-11.0.0.rc7/lib/active_fedora/solr_service.rb:45:in `query'
>        /usr/local/lib/ruby/gems/2.2.0/gems/curation_concerns-1.4.0/app/presenters/curation_concerns/work_show_presenter.rb:123:in `ordered_ids'
>        /usr/local/lib/ruby/gems/2.2.0/gems/curation_concerns-1.4.0/app/presenters/curation_concerns/work_show_presenter.rb:70:in `member_presenters'
>        /home/ubuntu/src/sufia/spec/presenters/sufia/work_show_presenter_spec.rb:76:in `block (3 levels) in <top (required)>'
>        /usr/local/lib/ruby/gems/2.2.0/gems/rspec-core-3.5.2/lib/rspec/core/example.rb:254:in `instance_exec'
>        /usr/local/lib/ruby/gems/2.2.0/gems/rspec-core-3.5.2/lib/rspec/core/example.rb:254:in `block in run'
>      # (eval):2:in `get'
>      # ./spec/presenters/sufia/work_show_presenter_spec.rb:76:in `block (3 levels) in <top (required)>'
> 
>

### CODE REVIEW

**Comment by jcoyne** [comment]
@hackmastera it looks like `id` is `nil` at `curation_concerns-a5b3dbbc1466/app/presenters/curation_concerns/work_show_presenter.rb:123`

File: spec/presenters/sufia/work_show_presenter_spec.rb [hunk 1]
```diff
@@ -67,4 +67,14 @@
       it { is_expected.to be true }
     end
   end
+
+  describe "#member_presenters" do
+    let(:work) { build(:work_with_one_file) }
+    let(:ability) { double "Ability" }
+
+    it "returns appropriate classes for each" do
+      expect(presenter.member_presenters.size).to eq 1
```
Comment (jcoyne) [inline spec/presenters/sufia/work_show_presenter_spec.rb]: `presenter.id` is `nil` here, that's the problem.

File: spec/presenters/sufia/work_show_presenter_spec.rb [hunk 2]
```diff
@@ -67,4 +67,14 @@
       it { is_expected.to be true }
     end
   end
+
+  describe "#member_presenters" do
+    let(:work) { build(:work_with_one_file) }
```
Comment (jcoyne) [inline spec/presenters/sufia/work_show_presenter_spec.rb]: This must be `create` instead of `build`, or the work doesn't have an id.

**Comment by hackmastera** [comment]
thank you for finding that, @jcoyne

**Comment by hackmastera** [comment]
this is now failing in the way I'm observing in the application:

> .............F
> 
> Failures:
> 
>   1) Sufia::WorkShowPresenter#member_presenters returns appropriate classes for each
>      Failure/Error: expect(presenter.member_presenters.first).to be_instance_of(::Sufia::FileSetPresenter)
>        expected #<CurationConcerns::FileSetPresenter:0x007fe96b78e8b0 @solr_document=#<SolrDocument:0x007fe96b78f288 ...+json"}, :ttl=>{:content_type=>"text/turtle"}}>, @current_ability=#<Double "Abil
> ity">, @request=nil> to be an instance of Sufia::FileSetPresenter
>      # ./spec/presenters/sufia/work_show_presenter_spec.rb:77:in `block (3 levels) in <top (required)>'

**Pull request closed without merging**